lib/puppet/functions/json_data.rb

```diff
@@ -0,0 +1,18 @@
+# @since 4.8.0
+#
+Puppet::Functions.create_function(:json_data) do
+  dispatch :json_data do
+    param 'Struct[{path=>String[1]}]', :options
+    param 'Puppet::LookupContext', :context
+  end
```

Comment:
This particular function and it's yaml counterpart, both take exactly one option, the path. All other options (including a misspelled paht) are incorrect. So why allow "free form" options here?